segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -124,7 +133,7 @@
         this.durableLog = durableLogFactory.createDurableLog(this.metadata, this.readIndex);
         shutdownWhenStopped(this.durableLog, "DurableLog");
         this.attributeIndex = attributeIndexFactory.createContainerAttributeIndex(this.metadata, this.storage, this.durableLog);
-        this.writer = writerFactory.createWriter(this.metadata, this.durableLog, this.readIndex, this.attributeIndex, this.storage);
+        this.writer = writerFactory.createWriter(this.metadata, this.durableLog, this.readIndex, this.attributeIndex, this.storage, this::createWriterProcessors);
         shutdownWhenStopped(this.writer, "Writer");
```

Comment:
I see that in this constructor `createWriterProcessors` is invoked, which internally iterates over `this.extensions`. Then, few lines after in the constructor `this.extensions` is initialized. Is this correct?

Reply:
It's not invoked, we pass the method reference as a lambda (look at the `::`). That will be invoked when the Writer itself is initialized (when `doStart` on this class is called), which is guaranteed to be after the constructor is done.